cmd/rule-evaluator/main.go

```diff
@@ -855,59 +901,52 @@ func (db *queryAccess) Close() error {
 
 // makeInstrumentedRoundTripper instruments the original RoundTripper with middleware to observe the request result.
 // The new RoundTripper counts the number of query requests sent to GCM and measures the latency of each request.
-func makeInstrumentedRoundTripper(transport http.RoundTripper, reg prometheus.Registerer) http.RoundTripper {
-	queryCounter := prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "rule_evaluator_query_requests_total",
-			Help: "A counter for query requests sent to GCM.",
-		},
-		[]string{"code", "method"},
-	)
-	queryHistogram := prometheus.NewHistogramVec(
-		prometheus.HistogramOpts{
-			Name:    "rule_evaluator_query_requests_latency_seconds",
-			Help:    "Histogram of response latency of query requests sent to GCM.",
-			Buckets: prometheus.DefBuckets,
-		},
-		[]string{"code", "method"},
-	)
-	reg.MustRegister(queryCounter, queryHistogram)
-
+func makeInstrumentedRoundTripper(transport http.RoundTripper) http.RoundTripper {
 	return promhttp.InstrumentRoundTripperCounter(queryCounter,
 		promhttp.InstrumentRoundTripperDuration(queryHistogram, transport))
 }
 
 type ruleEvaluator struct {
-	queryFunc    rules.QueryFunc
-	rulesManager *rules.Manager
+	queryFunc         rules.QueryFunc
+	rulesManager      *rules.Manager
+	lastEvaluatorOpts *evaluatorOptions
+}
+
+// Returns the URL that points to the rule-evaluator instance. By default, or if using a Google
+// Cloud base URL, this returns a link to the Google Cloud project page.
+func getExternalURL(generatorURL *url.URL, projectID string) *url.URL {
+	externalURL := generatorURL
+	if externalURL != nil && externalURL.String() == googleCloudBaseURL.String() {
+		// Project ID is empty when the rule-evaluator is instantiated, before config-reloader runs.
+		if projectID == "" {
+			externalURL = nil
+		} else {
+			// If it's a GCM link (default), create the full URL for the alert.
+			externalURL = googleCloudLink(projectID, "", time.Time{}, time.Time{})
+		}
+	}
+	return externalURL
```

Comment:
Want to reverse a bit for readability?

```suggestion
        // Project ID is empty when the rule-evaluator is instantiated, before config-reloader runs.
	if externalURL == nil || projectID == ""  {
	     return nil
	}
	
	if externalURL.String() == googleCloudBaseURL.String() {
		// If it's a GCM link (default), create the full URL for the alert.
		externalURL = googleCloudLink(projectID, "", time.Time{}, time.Time{})
	}
	return externalURL
```